wallet-core/src/keys.rs

```diff
@@ -40,6 +43,23 @@ pub fn derive_phoenix_sk(seed: &[u8; RNG_SEED], index: u8) -> PhoenixSecretKey {
     PhoenixSecretKey::random(&mut rng_with_index(seed, index, b"PSK"))
 }
 
+/// Generates multiple [`PhoenixSecretKey`] from a seed and a range of indices.
+///
+/// The randomness is generated using [`rng_with_index`].
+#[must_use]
+pub fn derive_multiple_phoenix_sk(
+    seed: &[u8; RNG_SEED],
+    index_range: Range<u8>,
+) -> Vec<PhoenixSecretKey> {
+    let mut keys = Vec::new();
+
+    for index in index_range {
+        keys.push(derive_phoenix_sk(seed, index));
+    }
+
+    keys
```

Comment:
While this is not wrong, I would probably avoid to use a temp mut var `keys` to collect all the keys:
```rust
pub fn derive_multiple_phoenix_sk(
    seed: &[u8; RNG_SEED],
    index_range: Range<u8>,
) -> Vec<PhoenixSecretKey> {
    index_range
        .map(|index| derive_phoenix_sk(seed, index))
        .collect()
}